addons/test_mail/tests/test_mail_channel.py

```diff
@@ -245,6 +245,24 @@ def test_channel_auto_unsubscribe_archived_or_deleted_users(self):
         self.assertEqual(test_channel_group.channel_partner_ids, self.env['res.partner'])
         self.assertEqual(self.test_channel.channel_partner_ids, self.user_employee.partner_id | test_partner)
 
+    def test_multi_company_chat(self):
+        company_A = self.env['res.company'].create({'name': 'Company A'})
+
+        test_user = self.env['res.users'].create({
+            'login': 'nomail',
+            'name': 'No Mail',
+            'company_ids': [(6, 0, company_A.ids)],
+            'company_id': company_A.id
+        })
+        test_chat = self.env['mail.channel'].with_context(self._test_context).create({
+            'channel_partner_ids': [(4, test_user.partner_id.id), (4, self.user_admin.partner_id.id)],
+            'public': 'private',
+            'channel_type': 'chat',
+            'name': 'test'
+        })
+        initial_channel_info = self.env['mail.channel'].with_context(allowed_company_ids=self.user_admin.company_ids.ids).channel_get(test_user.partner_id.ids)
```

Comment:
Finally a question here, do you know why you need to set `allowed_company_ids` in context here? Why is it not done automatically?

In the PR that fixed the bug, the request was directly coming from the JS, but someone/somewhere must have set that value in the context at some point for the issue to exist, do you know where?

Reply:
While I try to chat with multi-company users from UI side, it was passing the context of allowed company ids as a current company in channel_get.
and if I don't pass anything in the context it will be blank and the test was successful after also the previous fix removed.
so I have to pass the context here